frame/oracle/src/lib.rs

```diff
@@ -944,20 +959,27 @@ pub mod pallet {
 		#[allow(clippy::indexing_slicing)] // to get CI to pass
 		pub fn get_twap(
 			asset_id: T::AssetId,
-			mut price_weights: Vec<T::PriceValue>,
+			price_weights: Vec<Percent>,
```

Comment:
Why are the weights being passed from outside?

Reply:
I don't see the reason why this function should accept weights externally, but it's ok as it already exists.